### ISSUE TITLE
aten.matmul inefficient lowering for batched case

### ISSUE THREAD

**Issue by silvasean** [llvm/torch-mlir#879]
https://github.com/llvm/torch-mlir/blob/06750815d14371a7b8f0e964f2380517583e745f/lib/Conversion/TorchToLinalg/Linear.cpp#L242

We should implement this with a collapsing reshape, batch matmul, and then expanding reshape.

cc @ThomasRaoux

**Comment by pashu123** [comment]
https://github.com/llvm/torch-mlir/pull/792/files

**Comment by pashu123** [comment]
This works for the static case. The expanding reshape won't work for dynamic shapes, right @silvasean?

**Comment by powderluv** [comment]
Can we revive and merge the WIP? Or is there anything else blocking ?

**Comment by silvasean** [comment]
I don't think we should special case static shapes. We should do like we do for elementwise broadcasting ops and emit error checks for the "size-1" cases.

**Comment by silvasean** [comment]
Was this ever resolved?

**Comment by pashu123** [comment]
@silvasean This PR https://github.com/llvm/torch-mlir/pull/951/files handles this.

**Comment by silvasean** [comment]
Got it, closing.
